lib/api/model/submessage.dart

```diff
@@ -0,0 +1,285 @@
+import 'dart:convert';
+
+import 'package:json_annotation/json_annotation.dart';
+
+part 'submessage.g.dart';
+
+/// Data used for certain experimental Zulip widgets including polls and todo
+/// lists.
+///
+/// See:
+///   https://zulip.com/api/get-messages#response
+///   https://zulip.readthedocs.io/en/stable/subsystems/widgets.html
+@JsonSerializable(fieldRename: FieldRename.snake)
+class Submessage {
+  const Submessage({
+    required this.msgType,
+    required this.content,
+    required this.messageId,
+    required this.senderId,
+    required this.id,
+  });
+
+  @JsonKey(unknownEnumValue: SubmessageType.unknown)
+  final SubmessageType msgType;
+  /// [SubmessageData] encoded in JSON.
+  @JsonKey(readValue: readContent, toJson: contentToJson)
+  final Object? content;
+  final int messageId;
+  final int senderId;
+  final int id;
```

Comment:
Does this get used? I think not — so in that case let's ignore it, like `messageId` above.